tools/ui-components/src/form-control/form-control.tsx

```diff
@@ -2,58 +2,49 @@ import React from 'react';
 
 import { FormControlFeedback as Feedback } from './form-control-feedback';
 import { FormControlStatic as Static } from './form-control-static';
-import { FormControlProps, FormControlElement } from './types';
+import { FormControlProps } from './types';
 
 // Uses controlId from <FormGroup> if not explicitly specified.
 // type Only relevant if componentClass is 'input'.
 
-const FormControl = React.forwardRef<FormControlElement, FormControlProps>(
-  (
-    {
-      className,
-      id,
-      testId,
-      onChange,
-      value,
-      // eslint-disable-next-line @typescript-eslint/naming-convention
-      componentClass: Component = 'input',
-      placeholder,
-      name,
-      required,
-      type
-    },
-    ref
-  ): JSX.Element => {
-    let defaultClasses =
-      'outline-0 block w-full py-1.5 px-2.5 text-md text-foreground-primary ' +
-      'bg-background-primary bg-none rounded-none border-1 border-solid ' +
-      'border-background-quaternary shadow-none ' +
-      'transition ease-in-out duration-150 focus:border-foreground-tertiary';
+const FormControl = ({
+  className,
+  testId,
+  onChange,
+  value,
+  componentClass,
+  placeholder,
+  name,
+  required,
+  type,
+  ...restProps
+}: FormControlProps): JSX.Element => {
+  const defaultClasses =
+    'outline-0 block w-full py-1.5 px-2.5 text-md text-foreground-primary ' +
+    'bg-background-primary bg-none rounded-none border-1 border-solid ' +
+    'border-background-quaternary shadow-none ' +
+    'transition ease-in-out duration-150 focus:border-foreground-tertiary';
+  const Component = componentClass || 'input';
+  let variantClass;
+  if (Component !== 'textarea') variantClass = 'h-8';
```

Comment:
remove the `h-auto` assignment, because that is the default, and there is no value changing the height default value.

Reply:
If it is the default for the input and the text area, feel free to remove the condition.

Reply:
textarea doesn't have `h-8`, but does have `h-auto` as a default value.

Removing the condition will apply `h-8`, if it's `textarea`, which shouldn't be the case. what am I missing?

Reply:
h-auto is not in the defaultClasses. So the original logic makes sense.
You could change the 'h-8' to ' h-8' (with a space) so it gets parsed as a separate string.

Reply:
Every element created in the DOM has `h-auto` and `w-auto` as default values, except a few edge cases.

It is not existing doesn't change that textarea will have `h-auto`, because that is its default value. 


```suggestion
  if (Component !== 'textarea') variantClass = ' h-8';
```

Reply:
Sure, we will be overwriting browser defaults. That being said, we should still be alright.